packages/icons-workflow/bin/build-icons-mapping.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+import { getComponentName } from './build.js';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+// builds and saves the icons mapping to a json file
+const saveIconsMapping = async () => {
+    // check if there is file with .csv extension in the bin folder
+    const files = fs.readdirSync(path.join(__dirname, './'));
+    let data = '';
+    files.forEach((file) => {
+        if (file.endsWith('.csv')) {
+            data = fs.readFileSync(path.join(__dirname, file), 'utf8');
+        }
+    });
+
+    if (!data) {
+        throw new Error(
+            'No csv file found in the src folder. Please make sure to have the csv file in the bin folder'
+        );
+    }
+
+    const rows = data.split('\n');
+    const mapping = {};
+    rows.forEach((row) => {
+        // if the third column says "name change"
+        if (row.split(',')[2] === 'name change') {
```

Comment:
Could we start by moving the definition `const columns = row.split(',');` up?

nit: also if we were to use `for (const row of rows)` here, we could control the flow nicely, for example continuing `if (columns.length < 3 || !columns[2])`. But I am OK with the current approach!